cirq-web/cirq_ts/src/circuit/components/grid_qubit.ts

```diff
@@ -0,0 +1,112 @@
+// Copyright 2021 The Cirq Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Group, Vector3} from 'three';
+import {
+  QubitLine,
+  QubitLabel,
+  X3DSymbol,
+  BoxGate3DSymbol,
+  Control3DSymbol,
+  ConnectionLine,
+} from './meshes';
+
+/**
+ * Class that represents a GridQubit.
+ * A GridQubit consists of a three.js line, a sprite
+ * with location information, and three.js mesh objects
+ * representing the gates, if applicable.
+ */
+export class GridQubit extends Group {
+  readonly row: number;
+  readonly col: number;
+  readonly moments: number;
+
+  /**
+   * Class constructor.
+   * @param row The row of the GridQubit
+   * @param col The column of the GridQubit
+   * @param moments The number of moments of the entire circuit. This
+   * determines the length of the three.js line representing the GridQubit
+   */
+  constructor(row: number, col: number, moments: number) {
+    super();
+
+    this.row = row;
+    this.col = col;
+    this.moments = moments;
+    this.createLine();
+    this.addLocationLabel();
+  }
+
+  /**
+   * Adds a three.js mesh single qubit gate symbol
+   * to the group at the designated moment.
+   * @param label The label of the gate
+   * @param color The color of the gate
+   * @param moment The moment at which the gate occurs
+   */
+  addSingleQubitGate(label: string, color: string, moment: number) {
+    if (label === 'X') {
```

Comment:
I think the logic mapping a label to a symbol should belong to somewhere else, it is not the responsibility of the GridQubit to know about this mapping but rather it would be the duty of a Gate or SymbolRenderer class. GridQubit would have an `addSymbol` method instead that takes a `symbol` and a `moment`.